### PR TITLE
Fix MYSQL_ALLOW_EMPTY_PASSWORD check

### PR DESCRIPTION
**Pull request by vincentchalamon**
Fix https://github.com/docker-library/mysql/issues/92

### CODE REVIEW

**Comment by MLKiiwy** [comment]
:+1:

**Comment by yosifkit** [comment]
LGTM

File: 5.6/docker-entrypoint.sh [hunk 1]
```diff
@@ -48,14 +48,17 @@ if [ "$1" = 'mysqld' ]; then
...
+
 			DELETE FROM mysql.user ;
 			CREATE USER 'root'@'%' IDENTIFIED BY '${MYSQL_ROOT_PASSWORD}' ;
 			GRANT ALL ON *.* TO 'root'@'%' WITH GRANT OPTION ;
 			DROP DATABASE IF EXISTS test ;
 			FLUSH PRIVILEGES ;
 		EOSQL
-		mysql+=( -p"${MYSQL_ROOT_PASSWORD}" )
+
+		if [ -z "$MYSQL_ALLOW_EMPTY_PASSWORD" ]; then
```
Comment (tianon) [inline 5.6/docker-entrypoint.sh]: This needs to just be `-z "$MYSQL_ROOT_PASSWORD"` or else we'll fail on things like `-e MYSQL_ALLOW_EMPTY_PASSWORD=1 -e MYSQL_ROOT_PASSWORD=something` (which currently works).

We already bail above on `MYSQL_ALLOW_EMPTY_PASSWORD`, so at this point if `MYSQL_ROOT_PASSWORD` is empty, we have an empty password.

**Comment by guillaumevoisin** [comment]
:+1:

**Comment by ltangvald** [comment]
Teach me to look at the issues/prs before doing work. Was about to submit this exact PR, so LGTM :)

**Pull request merged**